### PR TITLE
renderer/gles: Drop shadow buffer

### PR DESCRIPTION
**Pull request by Drakulix**
As discussed previously, lets drop the shadow buffer code. We can scanout to 10-bit without it and any color-aware rendering is not going to end up in this renderer, so it just hurts performance. Which is especially bad considering this is supposed to be the most compatible renderer (apart from software rendering).

Also fixes https://github.com/Smithay/smithay/issues/1378

### CODE REVIEW

File: anvil/src/udev.rs [hunk 1]
```diff
@@ -244,14 +244,7 @@ pub fn run_udev() {
...
         let egl_context = EGLContext::new_with_priority(egl_display, ContextPriority::High)?;
-        let disable_color_transform = std::env::var("ANVIL_DISABLE_COLOR_TRANSFORM").is_ok();
-        unsafe {
-            let capabilities = GlesRenderer::supported_capabilities(&egl_context)?
-                .into_iter()
-                .filter(|c| !disable_color_transform || *c != Capability::ColorTransformations);
-
-            Ok(GlesRenderer::with_capabilities(egl_context, capabilities)?)
-        }
+        unsafe { Ok(GlesRenderer::new(egl_context)?) }
```
Comment (cmeissl) [inline anvil/src/udev.rs:247]: This could now use `GbmGlesBackend::with_context_priority(ContextPriority::High)`

File: anvil/src/udev.rs [hunk 2]
```diff
@@ -244,14 +244,7 @@ pub fn run_udev() {
 
     let gpus = GpuManager::new(GbmGlesBackend::with_factory(|egl_display| {
         let egl_context = EGLContext::new_with_priority(egl_display, ContextPriority::High)?;
-        let disable_color_transform = std::env::var("ANVIL_DISABLE_COLOR_TRANSFORM").is_ok();
```
Comment (cmeissl) [inline anvil/src/udev.rs:247]: The `README` of anvil also mentions  `ANVIL_DISABLE_COLOR_TRANSFORM`

**Review by cmeissl — CHANGES_REQUESTED** [review]

File: anvil/src/udev.rs (hunk 1 above)
Comment (Drakulix) [reply anvil/src/udev.rs:247]: Good call, fixed.

File: anvil/src/udev.rs (hunk 2 above)
Comment (Drakulix) [reply anvil/src/udev.rs:247]: Dropped from the readme.

**Review by cmeissl — APPROVED** [review]

**Comment by ids1024** [comment]
The shadow buffer code stood out on profiling, so this seems like a good change.

**Pull request merged**